### ISSUE TITLE
Cupertino tab_scaffold cannot modify currentPage for tabBar

### ISSUE THREAD

**Issue by paula23andrea** [flutter/flutter#31513]
### **Desired Situation**

If [cupertinoTabScaffoldState](https://github.com/flutter/flutter/blob/661e0350fbf302ed99495d0b4ff5137ebfd6737f/packages/flutter/lib/src/cupertino/tab_scaffold.dart#L161) were public and [currentPage](https://github.com/flutter/flutter/blob/661e0350fbf302ed99495d0b4ff5137ebfd6737f/packages/flutter/lib/src/cupertino/tab_scaffold.dart#L165) property too, It could change the actively displayed tab of the CupertinoTabScaffold from an child or acenstor through a GlobalKey.

```dart
static final cupertinoTabScaffoldStateKey = new GlobalKey<CupertinoTabScaffoldState>();

```

It could be useful when you need a "redirect" just changing the active tab.

For example, this code could be used for change the tab from everywhere:
```dart

MyKeys.cupertinoTabScaffoldStateKey.currentState.setState(()=>{ 
      MyKeys.cupertinoTabScaffoldStateKey.currentState.currentPage = myNewIndex 
});

```
My actual CupertinoTabScaffold

```dart
Map<String, WidgetBuilder> getRoutes(context, store) {
  return {
    '/': (BuildContext context) {
      return Scaffold(
        body: CupertinoTabScaffold(
          key: MyKeys.cupertinoTabScaffoldStateKey,
          tabBuilder: (BuildContext context, int index) {
            switch (index) {
              case 0: 
                return CupertinoTabView(
                  navigatorKey: MyKeys.homeNavigator,
                  routes: getRoutes(context, store),
                  onGenerateRoute: handleRoutes,
                  builder: (BuildContext context) => FeedScreen(),
                );
                break;
              case 1: 
                return CupertinoTabView(
                  navigatorKey: MyKeys.askNavigator,
                  routes: getRoutes(context, store),
                  onGenerateRoute: handleRoutes,
                  builder: (BuildContext context) => SearchResultScreen(arguments: {"askScreen": ""},),
                );
                break;
            }
          },
          tabBar: _buildTabBar(),
        ),
      );
    },
```

### **Problem**

Problem actually is the [cupertinoTabScaffoldState](https://github.com/flutter/flutter/blob/661e0350fbf302ed99495d0b4ff5137ebfd6737f/packages/flutter/lib/src/cupertino/tab_scaffold.dart#L161) and [currentPage](https://github.com/flutter/flutter/blob/661e0350fbf302ed99495d0b4ff5137ebfd6737f/packages/flutter/lib/src/cupertino/tab_scaffold.dart#L165) are private, so could it be public properties for making an implementation as showed in example?
Is there another way for changing the current tab from a children widget?

**Comment by xster** [comment]
This is solved in https://github.com/flutter/flutter/pull/31227. Let me know if I understood the question correctly.

**Comment by paula23andrea** [comment]
> This is solved in #31227. Let me know if I understood the question correctly.

Thank you! 
How long it takes to be in stable channel?
